### ISSUE TITLE
Background Update 'risk status' not working anymore on Android

### ISSUE THREAD

**Issue by lukas-wresch** [corona-warn-app/cwa-app-android#4657]
## Avoid duplicates
* [X] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)
* [X] Bug is specific for Android only, for general issues / questions that apply to iOS and Android please raise them in the [documentation repository](https://github.com/corona-warn-app/cwa-documentation)
* [X] Bug is not already reported in another issue

## Technical details

- Device name: Motorola Moto
- Android version: 8.1.0
- App version: 2.15.1

## Describe the bug

Background App Refresh has not been working for over a week.
The app does not update the risk status anymore.
"Prioritized Background Activity" is enabled in settings.
App has been closed and reopened several times.

## Steps to reproduce the issue

I tried to reproduce this behavior with two different Motorola smartphones, but failed to do so.

## Expected behaviour

The app should automatically update the risk status after 24h.

## Possible Fix

- On app startup check if the risk status is older than 24h. If yes, update the risk status.
- When the user swipes to the bottom on the status panel, update the risk status as long as the last update was at least a few minutes/hours ago.
- Create a button to manually initiated an update of the risk status. UX: The button should only be visible if the last update is older than 24h.

**Comment by Ein-Tim** [comment]
You mean the app is also not updating if it's in foreground?

**Comment by dsarkar** [comment]
@lukas-wresch Thanks for the report. Could send a screenshot from the status tab of the app, please? Thanks, DS


---
Corona-Warn-App Open Source Team

**Comment by Jo-Achim** [comment]
I didn't understand:
> I tried to reproduce this behavior with two different Motorola smartphones, but failed to do so.

Same error on two Motorola phones?

Regardless: Has the _smartphone_ been restarted to test this error?

**Comment by grussy** [comment]
Same here. CWA version 2.16.2 last update of risk status was on 18.1.  (today 30.1.) App does not update even when in foreground. All energy and background settings checked. Device rebooted, nothing helps.

Incidence Information is up to date.

**Comment by grussy** [comment]
> Same here. CWA version 2.16.2 last update of risk status was on 18.1. (today 30.1.) App does not update even when in foreground. All energy and background settings checked. Device rebooted, nothing helps.
> 
> Incidence Information is up to date.
![signal-2022-01-30-210025](https://user-images.githubusercontent.com/3177107/151715583-0b3a0178-99e6-407d-b469-04db4d64bbcf.jpeg)

**Comment by Ein-Tim** [comment]
@grussy 

It seems like you are experiencing #4472.

**Comment by vaubaehn** [comment]
The reason for the "stuck" background update is most likely a mixture of these factors:
- extraordinary high numbers of infections across Europe, hence many people uploading their DiagnosisKeys, hence huge packages
- old device that needs some time to process downloading of DiagnosisKeys for matching, getting DiagnosisKeys files copied to Google Play Services, validation of DK packages' signatures and the matching itself

which will lead to timeouts in different steps of the processes. When the OS kills the CWA thread before another attempt can be started, then the whole process begins from start the next time the watchdog wakes up CWA.

What could be checked is, whether the logic of the download/ENF provision process could be changed to reduce the amount of DK packages that are processed, like suggested here: https://github.com/corona-warn-app/cwa-app-android/issues/4732#issuecomment-1021095635
(if possible, download and submit only the last hourly DK package to ENS)

**Comment by dryo** [comment]
I just noticed that I ran into the same issue. I opened CWA, I saw a spinning circle and a message stating, that risk status was being updated. After a few seconds the circle and the message disappeared, but the last update date was 2022-01-10 (3 weeks ago!).

I rebooted the phone as I read on the internet that this could help. CWA still showed 2022-01-10. I went to "Einstellungen > Google > COVID-19 Benachrichtigungen > Drei-Punkte Symbol > Überprüfung" but "Überprüfung" was greyed out. I searched the internet, found this bug report, looked at my phone again and now it's no longer greyed out. I can see that 3,3 Mio. IDs have been downloaded a few minutes ago.

Whatever the reason was:

**CWA should display a warning if the last update date is older than X days or if download failed more than Y times.**

I didn't notice the old date for more than 3 weeks!
